src/pybind/mgr/rgw/module.py

```diff
@@ -229,6 +243,80 @@ def _parse_rgw_specs(self, inbuf: str) -> List[RGWSpec]:
+                    'prefix': 'osd pool create',
+                    'pool': pool_name,
+                    'pool_type': pool_type,
+                }
+                if pool_attrs:
+                    for k, v in pool_attrs.items():
+                        cmd_dict[k] = v
+                self.check_mon_command(cmd_dict)
+            except MonCommandFailed as e:
+                raise PoolCreationError(f'RGW module failed to create pool {pool_name} '
+                                        f'of type {pool_type} with attrs [{pool_attrs}]: {str(e)}')
+            # enable the rgw application on the pool
+            try:
+                self.check_mon_command({
+                    'prefix': 'osd pool application enable',
+                    'pool': pool_name,
+                    'app': 'rgw',
+                })
+            except MonCommandFailed as e:
+                raise PoolCreationError(f'Failed enabling application "rgw" on pool {pool_name}: {str(e)}')
+
+        zone_name = spec.rgw_zone
+        for pool_suffix in [
+            'buckets.index',
+            'meta',
+            'log',
+            'control'
+        ]:
+            # TODO: add size?
+            non_data_pool_attrs: Dict[str, Union[str, List[str]]] = {
+                'pg-num': '16' if 'index' in pool_suffix else '8',
+            }
+            _pool_create_command(f'{zone_name}.rgw.{pool_suffix}', 'replicated', non_data_pool_attrs)
+
+        if spec.data_pool_attributes:
+            if spec.data_pool_attributes.get('type', 'ec') == 'ec':
+                # we need to create ec profile
+                ec_pool_kv_pairs = {}
+                for attr in ['k', 'm', 'pg_num', 'crush-device-class']:
+                    if attr in spec.data_pool_attributes:
+                        ec_pool_kv_pairs[attr] = spec.data_pool_attributes[attr]
+                profile_name = f'{zone_name}_zone_data_pool_ec_profile'
+                profile_attrs = [f'{k}={v}' for k, v in ec_pool_kv_pairs.items()]
+                cmd_dict: Dict[str, Union[str, List[str]]] = {
+                    'prefix': 'osd erasure-code-profile set',
+                    'name': profile_name,
+                }
+                if profile_attrs:
+                    cmd_dict['profile'] = profile_attrs
+                try:
+                    self.check_mon_command(cmd_dict)
+                except MonCommandFailed as e:
+                    raise PoolCreationError(f'RGW module failed to create ec profile {profile_name} '
+                                            f'with attrs {profile_attrs}: {str(e)}')
+                # now we can pass the ec profile into the pool create command
+                data_pool_attrs: Dict[str, Union[str, List[str]]] = {
+                    'erasure_code_profile': profile_name
+                }
+                if 'pg_num' in spec.data_pool_attributes:
+                    data_pool_attrs['pg_num'] = spec.data_pool_attributes['pg_num']
```

Comment:
Just as suggestion, maybe this can go into a proper function for better cod readability, so we end-up with something like:

```
                ec_attrs = ...
                profile_name = _create_ec_profile(zone_name, ec_attrs)
                data_pool_attrs['erasure_code_profile'] = profile_name
```